Update default form definitions for Geo

grantfitzsimmons opened the pull request:
Fixes #5311, https://github.com/specify/specify7/issues/5073, https://github.com/specify/specify7/issues/3120, https://github.com/specify/specify7/issues/2153, https://github.com/specify/specify7/issues/1515, https://github.com/specify/specify7/issues/584, https://github.com/specify/specify7/issues/364, https://github.com/specify/specify6/issues/1210, https://github.com/specify/specify6/issues/1213, https://github.com/specify/specify6/issues/1229, https://github.com/specify/specify6/issues/1232, https://github.com/specify/specify7/issues/4591, https://github.com/specify/specify7/issues/5020, https://github.com/specify/specify7/issues/2016 and https://github.com/specify/specify6/issues/1223

This PR is intended to update underlying app resources necessary for the `geology` discipline as well as improve other defaults where possible.

## Geo-focused Improvements

### `CollectionObject` form for Geology
<img width="1624" alt="image" src="https://github.com/user-attachments/assets/f792a42e-372f-4b62-b98f-81ed40648a5f">

<img width="207" alt="image" src="https://github.com/user-attachments/assets/12a53a4f-7bf3-4ef8-b2d2-2510e797d613">


### Common Views

#### `RelativeAge`
<img width="1164" alt="image" src="https://github.com/user-attachments/assets/84d34add-1aef-4316-84a8-b225f9dc3d3d">

#### Added `AbsoluteAge` form
<img width="1168" alt="image" src="https://github.com/user-attachments/assets/56378d6d-0ec9-4b58-b00f-5ec2479c9b84">

#### Added `TectonicUnit` form
<img width="1520" alt="image" src="https://github.com/user-attachments/assets/e0beb81b-464c-4654-8de5-8b0acecbd08e">


#### Added `TectonicUnitTreeDef` form

<img width="1554" alt="image" src="https://github.com/user-attachments/assets/24205f9a-444b-40c3-a6be-564ace445670">


#### Added `TectonicUnitTreeDefItem` form
<img width="608" alt="image" src="https://github.com/user-attachments/assets/7576ed3a-f560-4073-92cc-76cc370f5e4c">


## Major Improvements

### Type Searches
* You can now search for `Collection`, `Institution`, and `SpecifyUser`, improving a number of places around the app!
* Instructions for users modifying this file are now included in the app resource itself

### Record Sets

<img width="517" alt="image" src="https://github.com/user-attachments/assets/beafec15-5354-4dd9-b5a9-dc96de4d2a18">

* You can now transfer ownership of a record set to another user!

### Agent

https://github.com/user-attachments/assets/a8a6c222-2064-4fa5-8c1b-7ed9d9c556d2

* `Person` is now the default value (**huge improvement**)
* Only when the Agent is type `Group` will the `members` subview display
* Agent Identifiers now displays properly, and subview is included on the default form

### Taxon

https://github.com/user-attachments/assets/f29069dc-6229-411a-a2e6-13fd5964f529

* Form is now conditional, displaying `HybridParent` fields only when `IsHybrid` is checked

### Collection Relationships
#### Old form:
<img width="500" alt="image" src="https://github.com/user-attachments/assets/ad411c05-d914-4069-90cd-b9edc1532154">

#### New form:
<img width="500" alt="image" src="https://github.com/user-attachments/assets/e0feb7e3-0a64-43ab-8277-423e08bcb35b">

* Can now easily create a Collection Relationship via the form
* Using the updated TypeSearches, "Collection" can be searched and displayed properly

### App Resources

#### AppResource

<img width="692" alt="image" src="https://github.com/user-attachments/assets/006367f7-6cee-41a7-ab0d-2d4371d640e5">

#### Reports

<img width="1301" alt="image" src="https://github.com/user-attachments/assets/4a1fbcef-a799-45ab-9e5e-c3954bdeab26">

* You can now easily find who owns the query and what the query is named from the app resource editor for a given report or label!
* The Reports subview button only appears when the mimetype is `jrxml/label`
* New `SpReport` form for the first time


#### SpViewSetObj
<img width="690" alt="image" src="https://github.com/user-attachments/assets/956e7f0c-1ab2-46a9-b90c-deb728145674">


### Pick List
#### Old form:
<img width="300" alt="image" src="https://github.com/user-attachments/assets/47bdd528-0016-4916-8808-67c18ac86416">

#### New form:
<img width="300" alt="image" src="https://github.com/user-attachments/assets/c8f797ce-7951-4f10-b4fb-f9708dc39b7b">

* Removed redundant header
* Now users can remove the pick list limit by setting it to `-1`

### Tree Definition
#### Old Tree Definition form:
<img width="300" alt="image" src="https://github.com/user-attachments/assets/ab5b6ea1-3a1d-43d5-84a6-ccef65f01fa9">

#### New Tree Definition form:
<img width="500" alt="image" src="https://github.com/user-attachments/assets/b65a7712-8f38-42ca-ad17-a4afde917058">

* Can now change full name direction for all trees

## Miscellaneous Improvements
* Preparation `countAmt` can no longer be set to a negative number by default
* Default form definition files and `DataObjFormatters` have finally been reformatted by the built-in Specify 7 system
* Removed a number of invalid fields on all default forms
* Updated hard-coded field labels to use schema-assigned captions instead


<!--
### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add automated tests
- [ ] Add relevant issue to release milestone
-->

# Testing instructions

## _THIS MUST BE TESTED LOCALLY UNTIL THE TEST PANEL SUPPORTS STATIC FILES_

Okay, so this PR is big, but it is almost entirely XML changes. This touches a lot of the default forms, so I have a big ask– please test the following XML resources in Specify. You are likely to find issues that aren't related to this PR which I am happy to resolve, so report them. If it grows outside the scope of this, I'll open another and we'll resolve it in the future.

Remember to **delete any custom form definitions / app resources** before testing this as those will be used first.

- [ ] Create a new `DataObjFormatters` app resource and verify that all table formats and aggregations appear as expected
- [ ] Build queries and return the (formatted) and (aggregated) results of as many tables as you can, with a focus on new Geo tables. If a table format or aggregation is missing where it should be included, let me know. Some tables (e.g. `*property`, `*attribute`, `*groupjoin`) do not have a table format or aggregation defined as it is ambiguous as to which fields we should display.
- [ ] Create a new `TypeSearches` app resource and verify that it is working as expected. Delete this resource after it is populated as you should be using the default.
- [ ] Open the `CollectionRelType` form and search for a  collection. You should be able to type the name of a collection and select it. You will encounter https://github.com/specify/specify7/issues/4989, but that is not in the scope of this issue, and you can safely dismiss it.
- [ ] Create an `ExportFeed` app resource and verify that you can search for a user in the database by name. You should see the name appear correctly _and_ you should be able to search for a user in the query combo box without the :mag: QB function.
- [ ] Create a new `WebLinks` app resource and verify that all web links (~5) appear and work as expected. See [this guide](https://discourse.specifysoftware.org/t/add-links-from-agent-to-orcid-bionomia-wikidata-and-more/1985) for setting up a conditional form on `AgentIdentifier` to leverage those new web links.
- [ ] Verify that the `geology.views.xml` can be used as the basis of a new form (create a new form definition and select `Geology`). Let me know what you think. This is subjective to some extent, and I welcome suggested changes (especially via XML snippets).

* Test that these new view definitions (mostly shown in this issue's body) are displayed properly (compare to defaults on `production`):
   - [ ] `CollectionRelType`
      - [ ] Verify that you can create a Collection Relationship without the query builder (ignoring this [error](https://github.com/specify/specify7/issues/4989))
   - [ ] `Agent` should default to `Person` instead of `Organization` _finally!_
      - [ ] `Agent` conditional: If the agent is not a `Group`, the `Members` subview should not be shown.
   - [ ] `TectonicUnit`
   - [ ] `TectonicUnitTreeDef`
   - [ ]  `TectonicUnitTreeDefItem`
   - [ ] `TaxonTreeDef`
   - [ ] `Taxon`
      - [ ] `Taxon` conditional: Verify that when you check `IsHybrid` the fields `HybridParent1` and `HybridParent2` appear. When it is not checked, they should not appear.
   - [ ] `GeographyTreeDef`
   - [ ] `StroageTreeDef`
   - [ ] `LithoStratTreeDef`
   - [ ] `GeologicTimePeriodTreeDef`
   - [ ] `RelativeAge`
   - [ ] `AbsoluteAge`
   - [ ] `PickList`
      - [ ] Set the size limit to `-1` and verify that you can add any number of pick list items (>500– and you can use the workbench to add the first 500)
   - [ ] `PickListItem`
   - [ ] `SpAppResource` (click pencil next to the title of an app resource)
       - [ ] `SpAppResource` conditional (should show `Reports` subview button if the resource is a report/label)
           - [ ] `SpReport` form should show ownership, associated query, title, in a read-only view
   - [ ] `SpViewSetObj` (click pencil next to the title of a form definition)
   - [ ] `SpQuery` (click pencil next to a query title)
       - [ ] Verify that you can change the ownership of a query
   - [ ] `RecordSet` (click pencil next to a record set title)
       - [ ] Verify that you can change the ownership of a record set
   - [ ] `AgentIdentifier` – navigate directly to the form via the data entry menu or by modifying the URL. It should no longer be auto-generated

- [ ] Double-check that your testing covers all cases reported in #5311, https://github.com/specify/specify7/issues/5073, https://github.com/specify/specify7/issues/3120, https://github.com/specify/specify7/issues/2153, https://github.com/specify/specify7/issues/1515, https://github.com/specify/specify7/issues/584, https://github.com/specify/specify7/issues/364, https://github.com/specify/specify6/issues/1210, https://github.com/specify/specify6/issues/1213, https://github.com/specify/specify6/issues/1229, https://github.com/specify/specify6/issues/1232, https://github.com/specify/specify7/issues/4591, https://github.com/specify/specify7/issues/5020, https://github.com/specify/specify7/issues/2016, and https://github.com/specify/specify6/issues/1223

grantfitzsimmons commented:
>You may only link 10 issues.

I can only link 10 issues but I have many more to solve with this PR.

melton-jason commented:
Haven't gone through all of the changes yet, but really nice so far! 

---

> The ideal fix would be to modify the default views, but since that can not occur until we separate from Specify 6, thankfully there seems to be enough consistency in cases for which this issue can be resolved

From https://github.com/specify/specify7/pull/4760#issuecomment-2046022176

Now we can properly resolve this! 🥳 

I've left a full breakdown of the Issue in the aforementioned comment (https://github.com/specify/specify7/pull/4760#issuecomment-2046022176), but the important part in relation to this PR is the following: 

> In the default forms, it seems there are two separate ways to represent a FormTable.
>
> The first and expected way is the case like Preparation, in which the Grid and Form view definitions are defined in a single view.
>
> The other way which is present is to have two separate views, where each view and their view definitions separately represent the Grid and Form views in the form.
>
> Specifically, the table-only views contain viewdefs which are not listed as altviews of the view and are only referenced through the formtable viewdef.
>
> This causes a problem in Specify 7 which expects all view definitions relating to a view to be associated with the view and not split amongst multiple views.

It'd be great if we can go through and merge these separate views for the collapsed version of the Subview with the expanded "normal" views! 

We already have a list of viewnames in the following commit: 
https://github.com/specify/specify7/commit/5c91a89e633130899af46b3446aaa282ef41920d
(Which was recorded in https://github.com/specify/specify7/issues/4737#issuecomment-2048322097)

Which exist at the following locations: 

| View Name              | File Path                  |
|-----------------------|---------------------------|
| Authors               | backstop/global.views.xml |
| GroupPersons          | backstop/global.views.xml |
| Collectors            | common/common.views.xml   |
| CollectionObjectSubTable | common/common.views.xml   |
| ContainerBriefTable   | common/common.views.xml   |
| Determiners           | common/common.views.xml   |
| Extractors            | common/common.views.xml   |
| FundingAgents         | common/common.views.xml   |
| PcrPersons            | common/common.views.xml   |

Specifically: for each of these views, we need to add the `formtable` viewdefs to the `altviews` of the main table's view (specifically, the `table.view` as specified in the datamodel. Which is [almost?] always just the table name). 
Essentially, align the views with how a view like `Preparation` is defined. 

I can make these changes if needed!

Related lines of code: 

https://github.com/specify/specify7/blob/dee374c3544bfbc73c611d7900d3e98366fa0637/specifyweb/frontend/js_src/lib/components/FormParse/index.ts#L77-L86

https://github.com/specify/specify7/blob/dee374c3544bfbc73c611d7900d3e98366fa0637/specifyweb/frontend/js_src/lib/components/FormParse/index.ts#L153-L218

alesan99 reviewed (changes requested):
# Testing instructions

- [x] Create a new `DataObjFormatters` app resource and verify that all table formats and aggregations appear as expected
- [ ] Build queries and return the (formatted) and (aggregated) results of as many tables as you can, with a focus on new Geo tables. If a table format or aggregation is missing where it should be included, let me know. Some tables (e.g. `*property`, `*attribute`, `*groupjoin`) do not have a table format or aggregation defined as it is ambiguous as to which fields we should display.
- [x] Create a new `TypeSearches` app resource and verify that it is working as expected. Delete this resource after it is populated as you should be using the default.
- [x] Open the `CollectionRelType` form and search for a  collection. You should be able to type the name of a collection and select it. You will encounter https://github.com/specify/specify7/issues/4989, but that is not in the scope of this issue, and you can safely dismiss it.
- [x] Create an `ExportFeed` app resource and verify that you can search for a user in the database by name. You should see the name appear correctly _and_ you should be able to search for a user in the query combo box without the :mag: QB function.
- [x] Create a new `WebLinks` app resource and verify that all web links (~5) appear and work as expected. See [this guide](https://discourse.specifysoftware.org/t/add-links-from-agent-to-orcid-bionomia-wikidata-and-more/1985) for setting up a conditional form on `AgentIdentifier` to leverage those new web links.
- [x] Verify that the `geology.views.xml` can be used as the basis of a new form (create a new form definition and select `Geology`). Let me know what you think. This is subjective to some extent, and I welcome suggested changes (especially via XML snippets).

* Test that these new view definitions (mostly shown in this issue's body) are displayed properly (compare to defaults on `production`):
   - [x] `CollectionRelType`
      - [x] Verify that you can create a Collection Relationship without the query builder (ignoring this [error](https://github.com/specify/specify7/issues/4989))
   - [x] `Agent` should default to `Person` instead of `Organization` _finally!_
      - [x] `Agent` conditional: If the agent is not a `Group`, the `Members` subview should not be shown.
   - [x] `TectonicUnit`
   - [x] `TectonicUnitTreeDef`
   - [x]  `TectonicUnitTreeDefItem`
   - [x] `TaxonTreeDef`
   - [x] `Taxon`
      - [x] `Taxon` conditional: Verify that when you check `IsHybrid` the fields `HybridParent1` and `HybridParent2` appear. When it is not checked, they should not appear.
   - [x] `GeographyTreeDef`
   - [x] `StroageTreeDef`
   - [x] `LithoStratTreeDef`
   - [x] `GeologicTimePeriodTreeDef`
   - [x] `RelativeAge`
   - [x] `AbsoluteAge`
   - [x] `PickList`
      - [x] Set the size limit to `-1` and verify that you can add any number of pick list items (>500– and you can use the workbench to add the first 500)
   - [x] `PickListItem`
   - [x] `SpAppResource` (click pencil next to the title of an app resource)
       - [ ] `SpAppResource` conditional (should show `Reports` subview button if the resource is a report/label)
           - [ ] `SpReport` form should show ownership, associated query, title, in a read-only view
   - [x] `SpViewSetObj` (click pencil next to the title of a form definition)
   - [x] `SpQuery` (click pencil next to a query title)
       - [x] Verify that you can change the ownership of a query
   - [x] `RecordSet` (click pencil next to a record set title)
       - [ ] Verify that you can change the ownership of a record set
   - [x] `AgentIdentifier` – navigate directly to the form via the data entry menu or by modifying the URL. It should no longer be auto-generated

- [x] Double-check that your testing covers all cases reported

Looking good! A lot of longstanding issues are addressed 👍

Here's some things I found so far (not sure if all of these are issues):

- There's no formatter defined for Tectonic Units
![image](https://github.com/user-attachments/assets/16029d37-34e5-4532-87b8-f58226b5fa7c)
- The pick list form doesn't conditionally hide certain fields like max mentioned in https://github.com/specify/specify7/issues/364#issuecomment-1184721825. I think it would be nice, but its functional anyway
- I am not getting a report subview button for reports. Labels yes.
![chrome_6iGDkpnDxk](https://github.com/user-attachments/assets/df9883e2-b8de-40d2-8a98-caefa05afb4f)
- SpReport fields don't appear read-only
![chrome_1UAL13eB8Q](https://github.com/user-attachments/assets/63501dfb-a0f4-446e-a551-8e3322a50eee)
- I cannot change ownership of record set
![chrome_fEGIV87c19](https://github.com/user-attachments/assets/f72890a6-346f-4267-b5e1-94925a5bd227)

I can investigate these further to see if any of these are a mistake on my part

grantfitzsimmons commented:
Hi @alesan99 – thank you for your review!

> - There's no formatter defined for Tectonic Units
> ![image](https://github.com/user-attachments/assets/16029d37-34e5-4532-87b8-f58226b5fa7c)

Added TectonicUnit table format in [`cd54372`](https://github.com/specify/specify7/pull/5367/commits/cd5437212662288a4e5d5c91c7d14859320a7d22)

>  - [ ] `SpAppResource` conditional (should show `Reports` subview button if the resource is a report/label)
>      - [ ] `SpReport` form should show ownership, associated query, title, in a read-only view
> - SpReport fields don't appear read-only

Oh, good! I was hoping they would be editable. Not sure why they weren't in my database.

> - The pick list form doesn't conditionally hide certain fields like max mentioned in https://github.com/specify/specify7/issues/364#issuecomment-1184721825. I think it would be nice, but its functional anyway

I consciously decided against making this a conditional form, but it might be worth revisiting in the future. I didn't want to obscure data that was still captured in the database record.

> - I cannot change ownership of record set

Fixed in [`0995dfe`](https://github.com/specify/specify7/pull/5367/commits/0995dfef12ec06f5fcadf09a5973e3d854ac132a)

sharadsw commented on config/common/common.views.xml:
```diff
@@ -2634,32 +1944,55 @@
+							<row/>
+							<!-- Show nothing if no Collection Object Group (COG) or Collection Object (CO) child exists for this COJO record -->
+						</rows>
+						<rows condition="childCog.collection.ID=4">
```
Hardcoded id here as well?

grantfitzsimmons replied:
Fixed in [`1805b60` (#5367)](https://github.com/specify/specify7/pull/5367/commits/1805b60eab4cb7f20798dd561ff299e7eb1f0f71)

sharadsw commented on config/common/common.views.xml:
```diff
@@ -2598,34 +1910,32 @@
-					<cell type="subview" id="cojo" name="cojo" uitype="subview" isrequired="false" defaulttype="table" initialize="btn=true; align=center" colspan="13"/>
+					<!--<cell type="label" labelfor="parentCojo"/>
+					<cell type="field" id="parentCojo" name="parentCojo" uitype="querycbx" isrequired="false"/> -->
+					<cell type="subview" id="cojo" name="childCojos" uitype="subview" initialize="btn=true"/>
```
`childCojos` would have to be changed to `children`

grantfitzsimmons replied:
Fixed in [`a2108ab` (#5367)](https://github.com/specify/specify7/pull/5367/commits/a2108ab50e85e7053eac5ed6d6db865c2a3fa302)

sharadsw commented on config/common/common.views.xml:
```diff
@@ -2598,34 +1910,32 @@
-					<cell type="label" labelfor="parentCojo"/>
-					<cell type="field" id="parentCojo" name="parentCojo" uitype="querycbx" isrequired="false"/>
+					<cell type="field" id="cogType" name="cogType" uitype="combobox"/>
+					<cell type="label" label="Parent COG"/>
+					<cell type="field" id="parentCojo" name="parentCojo" uitype="querycbx"/>
```
`parentCojo` was removed in #5406, so this field would have to be changed to `parentCog`

grantfitzsimmons replied:
Fixed in https://github.com/specify/specify7/pull/5367/commits/5b150d5cdc3be2a2aeb3c38d43b3b839f5bbf4cd

sharadsw reviewed (commented):
Looks great! I only have some suggestions for COG-related viewdefs

grantfitzsimmons commented on config/common/common.views.xml:
```diff
@@ -2634,32 +1944,55 @@
 						</rows>
-						<rows condition="childCog.CollectionObjectGroupID=">
-							<!-- If there is no CollectionObjectGroupID, display CO -->
+						<rows condition="childCo.collection.ID=4">
```
Thank you!
If changes like this are made, we need to update the documentation. We have "lost" features before.

grantfitzsimmons replied:
Fixed in [`1805b60` (#5367)](https://github.com/specify/specify7/pull/5367/commits/1805b60eab4cb7f20798dd561ff299e7eb1f0f71)

alesan99 reviewed (changes requested):
All my issues were addressed 👍👍👌

I did also notice Tectonic Unit Tree Def and Tectonic Unit Tree Def Item are missing formatters, where their Taxon and Storage counterparts do have one.

![image](https://github.com/user-attachments/assets/c9146c06-d4d7-4aa9-a1c0-6ecdcfeec19f)
![image](https://github.com/user-attachments/assets/00a52063-15d0-4fc6-b679-91046ec980d4)


And yeah I see the COG form was missing Parent COG because of what Sharad mentioned.

alesan99 reviewed (approved):
Parent COG is appearing 👍

![image](https://github.com/user-attachments/assets/142c5660-ea96-405d-b16b-44488b739da7)

The new conditionals are also working

![image](https://github.com/user-attachments/assets/aa677ac7-0ce0-4eec-88a6-dca33fdc671e)

![image](https://github.com/user-attachments/assets/54a49f12-9736-4f3a-a85c-a4f113ba7a37)

pashiav reviewed (dismissed):
# Testing instructions
- [X] Create a new `DataObjFormatters` app resource and verify that all table formats and aggregations appear as expected
- [X] Build queries and return the (formatted) and (aggregated) results of as many tables as you can, with a focus on new Geo tables. If a table format or aggregation is missing where it should be included, let me know. Some tables (e.g. `*property`, `*attribute`, `*groupjoin`) do not have a table format or aggregation defined as it is ambiguous as to which fields we should display.
- [X] Create a new `TypeSearches` app resource and verify that it is working as expected. Delete this resource after it is populated as you should be using the default.
- [ ] Open the `CollectionRelType` form and search for a  collection. You should be able to type the name of a collection and select it. You will encounter https://github.com/specify/specify7/issues/4989, but that is not in the scope of this issue, and you can safely dismiss it.
- [X] Create an `ExportFeed` app resource and verify that you can search for a user in the database by name. You should see the name appear correctly _and_ you should be able to search for a user in the query combo box without the :mag: QB function.
- [X] Create a new `WebLinks` app resource and verify that all web links (~5) appear and work as expected. See [this guide](https://discourse.specifysoftware.org/t/add-links-from-agent-to-orcid-bionomia-wikidata-and-more/1985) for setting up a conditional form on `AgentIdentifier` to leverage those new web links.
- [X] Verify that the `geology.views.xml` can be used as the basis of a new form (create a new form definition and select `Geology`). Let me know what you think. This is subjective to some extent, and I welcome suggested changes (especially via XML snippets).

* Test that these new view definitions (mostly shown in this issue's body) are displayed properly (compare to defaults on `production`):
   - [ ] `CollectionRelType`
      - [ ] Verify that you can create a Collection Relationship without the query builder (ignoring this [error](https://github.com/specify/specify7/issues/4989))
   - [X] `Agent` should default to `Person` instead of `Organization` _finally!_
      - [X] `Agent` conditional: If the agent is not a `Group`, the `Members` subview should not be shown.
   - [X] `TectonicUnit`
   - [X] `TectonicUnitTreeDef`
   - [X]  `TectonicUnitTreeDefItem`
   - [X] `TaxonTreeDef`
   - [X] `Taxon`
      - [X] `Taxon` conditional: Verify that when you check `IsHybrid` the fields `HybridParent1` and `HybridParent2` appear. When it is not checked, they should not appear.
   - [X] `GeographyTreeDef`
   - [X] `StroageTreeDef`
   - [X] `LithoStratTreeDef`
   - [X] `GeologicTimePeriodTreeDef`
   - [X] `RelativeAge`
   - [X] `AbsoluteAge`
   - [X] `PickList`
      - [X] Set the size limit to `-1` and verify that you can add any number of pick list items (>500– and you can use the workbench to add the first 500)
   - [X] `PickListItem`
   - [ ] `SpAppResource` (click pencil next to the title of an app resource)
       - [ ] `SpAppResource` conditional (should show `Reports` subview button if the resource is a report/label)
           - [X] `SpReport` form should show ownership, associated query, title, in a read-only view
   - [X] `SpViewSetObj` (click pencil next to the title of a form definition)
   - [ ] `SpQuery` (click pencil next to a query title)
       - [ ] Verify that you can change the ownership of a query
   - [X] `RecordSet` (click pencil next to a record set title)
       - [X] Verify that you can change the ownership of a record set
   - [X] `AgentIdentifier` – navigate directly to the form via the data entry menu or by modifying the URL. It should no longer be auto-generated

- [X] Double-check that your testing covers all cases reported in #5311, https://github.com/specify/specify7/issues/5073, https://github.com/specify/specify7/issues/3120, https://github.com/specify/specify7/issues/2153, https://github.com/specify/specify7/issues/1515, https://github.com/specify/specify7/issues/584, https://github.com/specify/specify7/issues/364, https://github.com/specify/specify6/issues/1210, https://github.com/specify/specify6/issues/1213, https://github.com/specify/specify6/issues/1229, https://github.com/specify/specify6/issues/1232, https://github.com/specify/specify7/issues/4591, https://github.com/specify/specify7/issues/5020, https://github.com/specify/specify7/issues/2016, and https://github.com/specify/specify6/issues/1223

---
---
Issues/Comments
---
WOW looks soooo much better! 

Using db `ciscollections_2_15_24` to test.

**1.** 
> - [ ] Open the `CollectionRelType` form and search for a  collection. You should be able to type the name of a collection and select it. You will encounter https://github.com/specify/specify7/issues/4989, but that is not in the scope of this issue, and you can safely dismiss it.

&

>    - [ ] `CollectionRelType`
      - [ ] Verify that you can create a Collection Relationship without the query builder (ignoring this [error](https://github.com/specify/specify7/issues/4989))

No results pop up when I type in a name of a collection. I can only select the collection through query builder.

https://github.com/user-attachments/assets/e8282e13-85ac-4ea5-a502-118ca6fc71e2

<br>

**2.** 

> - [X] Verify that the `geology.views.xml` can be used as the basis of a new form (create a new form definition and select `Geology`). Let me know what you think. This is subjective to some extent, and I welcome suggested changes (especially via XML snippets).

I think it's a great base to work off of! It's simple and looks nice.

<br>

**3.**

> - [ ] `SpAppResource` conditional (should show `Reports` subview button if the resource is a report/label)

The `Reports` subview button shows up for Label resources, but does not for Report resources.

https://github.com/user-attachments/assets/c8126647-dc50-45da-bf74-a33d37b33085

<br>

**4.**

> - [X] `SpReport` form should show ownership, associated query, title, in a read-only view

Also is not read-only on my db!


**5.**
>  - [ ] Verify that you can change the ownership of a query

I can change ownership, but only through query builder. I cannot select users by searching their name in the field.

https://github.com/user-attachments/assets/969b8386-8c87-4b1e-88ba-2acdcd93c2e5

<br>

See that I can change the ownership of a record set through searching in the field:

https://github.com/user-attachments/assets/60cf041b-159b-4735-9905-747ae941e724

combs-a reviewed (dismissed):
# Testing instructions

## _THIS MUST BE TESTED LOCALLY UNTIL THE TEST PANEL SUPPORTS STATIC FILES_

Okay, so this PR is big, but it is almost entirely XML changes. This touches a lot of the default forms, so I have a big ask– please test the following XML resources in Specify. You are likely to find issues that aren't related to this PR which I am happy to resolve, so report them. If it grows outside the scope of this, I'll open another and we'll resolve it in the future.

Remember to **delete any custom form definitions / app resources** before testing this as those will be used first.

- [ ] Create a new `DataObjFormatters` app resource and verify that all table formats and aggregations appear as expected
- [ ] Build queries and return the (formatted) and (aggregated) results of as many tables as you can, with a focus on new Geo tables. If a table format or aggregation is missing where it should be included, let me know. Some tables (e.g. `*property`, `*attribute`, `*groupjoin`) do not have a table format or aggregation defined as it is ambiguous as to which fields we should display.
- [X] Create a new `TypeSearches` app resource and verify that it is working as expected. Delete this resource after it is populated as you should be using the default.
- [X] Open the `CollectionRelType` form and search for a  collection. You should be able to type the name of a collection and select it. You will encounter https://github.com/specify/specify7/issues/4989, but that is not in the scope of this issue, and you can safely dismiss it.
- [X] Create an `ExportFeed` app resource and verify that you can search for a user in the database by name. You should see the name appear correctly _and_ you should be able to search for a user in the query combo box without the :mag: QB function.
- [X] Create a new `WebLinks` app resource and verify that all web links (~5) appear and work as expected. See [this guide](https://discourse.specifysoftware.org/t/add-links-from-agent-to-orcid-bionomia-wikidata-and-more/1985) for setting up a conditional form on `AgentIdentifier` to leverage those new web links.
- [X] Verify that the `geology.views.xml` can be used as the basis of a new form (create a new form definition and select `Geology`). Let me know what you think. This is subjective to some extent, and I welcome suggested changes (especially via XML snippets).

* Test that these new view definitions (mostly shown in this issue's body) are displayed properly (compare to defaults on `production`):
   - [X] `CollectionRelType`
      - [X] Verify that you can create a Collection Relationship without the query builder (ignoring this [error](https://github.com/specify/specify7/issues/4989))
   - [X] `Agent` should default to `Person` instead of `Organization` _finally!_
      - [X] `Agent` conditional: If the agent is not a `Group`, the `Members` subview should not be shown.
   - [X] `TectonicUnit`
   - [X] `TectonicUnitTreeDef`
   - [X]  `TectonicUnitTreeDefItem`
   - [X] `TaxonTreeDef`
   - [X] `Taxon`
      - [X] `Taxon` conditional: Verify that when you check `IsHybrid` the fields `HybridParent1` and `HybridParent2` appear. When it is not checked, they should not appear.
   - [X] `GeographyTreeDef`
   - [X] `StroageTreeDef`
   - [X] `LithoStratTreeDef`
   - [X] `GeologicTimePeriodTreeDef`
   - [X] `RelativeAge`
   - [X] `AbsoluteAge`
   - [X] `PickList`
      - [X] Set the size limit to `-1` and verify that you can add any number of pick list items (>500– and you can use the workbench to add the first 500)
   - [X] `PickListItem`
   - [X] `SpAppResource` (click pencil next to the title of an app resource)
       - [ ] `SpAppResource` conditional (should show `Reports` subview button if the resource is a report/label)
           - [ ] `SpReport` form should show ownership, associated query, title, in a read-only view
   - [X] `SpViewSetObj` (click pencil next to the title of a form definition)
   - [X] `SpQuery` (click pencil next to a query title)
       - [X] Verify that you can change the ownership of a query
   - [X] `RecordSet` (click pencil next to a record set title)
       - [X] Verify that you can change the ownership of a record set
   - [X] `AgentIdentifier` – navigate directly to the form via the data entry menu or by modifying the URL. It should no longer be auto-generated

- [ ] Double-check that your testing covers all cases reported in #5311, https://github.com/specify/specify7/issues/5073, https://github.com/specify/specify7/issues/3120, https://github.com/specify/specify7/issues/2153, https://github.com/specify/specify7/issues/1515, https://github.com/specify/specify7/issues/584, https://github.com/specify/specify7/issues/364, https://github.com/specify/specify6/issues/1210, https://github.com/specify/specify6/issues/1213, https://github.com/specify/specify6/issues/1229, https://github.com/specify/specify6/issues/1232, https://github.com/specify/specify7/issues/4591, https://github.com/specify/specify7/issues/5020, https://github.com/specify/specify7/issues/2016, and https://github.com/specify/specify6/issues/1223

## Tester Comments

TectonicUnit was being rude but I'm pretty sure it's just a database issue that I need to sort on my end. Wasn't able to open it in DataObjFormatters or the Query Builder, and it would give definitions missing errors + couldn't fetch ranks in the form. I can't really pass the first two checks because I can't check that, but I had a few other comments anyway, so I'll retest them both once I've got that figured out. 😅 

Reports button missing on the report SpAppResource. It does appear on the label, however.
![image](https://github.com/user-attachments/assets/3c6b674b-f0fa-4224-a6a8-6deca83508da)

Had an issue with report runner, so I wasn't able to check if it was read only in the labels/reports form. I was able to access it by modifying the link in Data Entry and it wasn't read only there, though.

For the record sets, having owner in the default form means that when you create a record set from query the user has to manually fill in the owner, which is a little annoying, but I assume it's outside the scope of the PR to have it autofill the current user as the owner. I do wonder who is able to transfer ownership to other people, though. There's no specific permissions in place to bar people from transferring right? It would have to be form-based?

I'm unsure about the hard-coding checks, since here on the Picklist form there's a few that look like they might be (Type, Name, Formatter), but that's just going based on my assumption that the field names are always lowercase.

![image](https://github.com/user-attachments/assets/6b2512b8-6431-4154-9064-5c48ee13ec1f)

That's all I found wrong with the last check, these new default forms are great!
